packages/types/src/signer.ts

```diff
@@ -0,0 +1,24 @@
+import { Account } from "./account";
+import {
+  IbcTransferProps,
+  InitAccountProps,
+  SignedTx,
+  TransferProps,
+  TxProps,
+} from "./tx";
+
+export interface Signer<T = Account> {
+  accounts: () => Promise<T[] | undefined>;
```

Comment:
1
```suggestion
export interface ISigner<T = Account> {
  accounts: () => Promise<T[] | undefined>;
```
It seems that we are aliasing this at a few imports as `ISigner` as `Signer` class is implementing it. As we have a class of same name it would make searching for their usage much easier to have different names for them.


2
Also I am wondering why do we not speak about just `Account[]` here instead of `T[]`. To me it seems that we are using this `interface Account` in 2 occasions:
* in one of them we use `Account` for the generic type
* in the other occasion we just omit it, so it defaults this this `Account`

Or are going to have some future use case where `T` will have to be something different than `Account`?

```suggestion
export interface Signer {
  accounts: () => Promise<Account | undefined>;
```

Reply:
Yeah we could probably just use `Account` and not have it be generic. I think I had originally thought that there may be additional types that could be used here, but I don't think a generic is necessary.